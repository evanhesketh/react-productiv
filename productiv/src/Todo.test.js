import React from "react";
import { render } from "@testing-library/react";
import Todo from "./Todo";

// let renderedTodo;

// beforeEach(function () {
//   renderedTodo = render(<Todo
//     title="Test todo"
//     description="Test description"
//     priority={1} />);
// });

describe("todo", function () {
  it("renders without crashing", function () {
    render(<Todo
      title="Test todo"
      description="Test description"
      priority={1} />);
  });

  it("matches snapshot on initial render", function () {
    const { container } = render(<Todo
      title="Test todo"
      description="Test description"
      priority={1} />);
    expect(container).toMatchSnapshot();
  });

  it("contains expected information", function () {
    const { container } = render(<Todo
      title="Test todo"
      description="Test description"
      priority={1} />);
    expect(container.querySelector(".Todo")).toContainHTML("Test todo");
  });


});