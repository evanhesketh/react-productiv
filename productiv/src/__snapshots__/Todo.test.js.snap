// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todo matches snapshot on initial render 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        Test todo
      </b>
       
      <small>
        Priority: 
        1
      </small>
    </div>
    <div>
      <small>
        Test description
      </small>
    </div>
  </div>
</div>
`;
